src/lib/include/simd/simd_floats.h

```diff
@@ -0,0 +1,158 @@
+#ifndef DISTOPIA_SIMD_FLOAT_H
+#define DISTOPIA_SIMD_FLOAT_H
+
+#include <cassert>
+#include <immintrin.h>
+
+// class to wrap __m128 4 x packed floats
+class SimdFloatX4 {
```

Comment:
Instead of implementing this structure four times, I suggest using templates. You can parametrize of the float kind (single- vs double-precision) and the vector width (128-, 256-, or 512-bit).

Reply:
yep yep, makes so much more sense.